packages/@lwc/template-compiler/src/parser/expression.ts

```diff
@@ -33,57 +33,62 @@ export function isPotentialExpression(source: string): boolean {
     return !!source.match(POTENTIAL_EXPRESSION_RE);
 }
 
+function validateExpression(
+    node: estree.BaseNode,
+    element: IRNode,
+    allowComputedMemberExpression: boolean
+): node is estree.MemberExpression | estree.Identifier {
+    const isValidNode = node.type === 'Identifier' || node.type === 'MemberExpression';
+    invariant(isValidNode, ParserDiagnostics.INVALID_NODE, [node.type]);
```

Comment:
We should create helpers to simplify this code. This would centralize the type checks in a single place and avoid having to use type guards in the code.

```ts
function isEsTreeIdentifer(node: estree.BaseNode): node is estree.Identifier {
    return node.type === 'Identifier';
}

function isEsTreeMemberExpression(node: estree.BaseNode): node is estree.MemberExpression {
    return node.type === 'MemberExpression';
}
```